extended-utxo-spec/extended-utxo-specification.tex

```diff
@@ -1016,6 +997,34 @@ \subsection{The EUTXO-2 model in practice.}
 \section{Comments on the specification}
 \label{appendix:comments}
 
+\note{Computing with finitely-supported functions}
+\label{note:finitely-supported-functions}
+
+We intend that finitely-supported functions are implemented as finite
+maps. However, there are two apparent difficulties:
+\begin{enumerate}
+  \item The domain of a map does not correspond to the support of the function:
+    values may be mapped to zero, thus appearing in the domain but not the support.
+  \item Pointwise equality is hard to compute.
+\end{enumerate}
+
+However, both of these are easily ameliorated. We say that a set $w$ is a \textit{weak support}
+of a finitely-supported function $f$ if $\support(f) \subseteq w$. That is, a
+weak support contains all the points that are non-zero, but possibly also some
+points that are zero. It is easy to see that the domain of a map is a weak
+support for the finitely-supported function it represents.
+
+We can compute the support from the weak support by simply checking the function
+at each value and removing those that are zero. This is potentially expensive,
+but we only need to do it when we need the support, which we only do during the
+computation of rule~\ref{rule:value-is-preserved-2}.
+
```

Comment:
No, it's different. It happens in a different place. In particular we don't do it during balancing, we only need to do it when we're checking the support for forging.

And yes, you need to do some work, but it's all described in the note. To find the support you go through the weak support and remove all values that are zero, and to check for zero you do pointwise comparison over the weak support.

Reply:
And a major point of doing this was so that the main text can just rely on the algebraic properties and primitives defined earlier, and we can put the operational content about how to implement the basics off to the side.